apps/emqx_management/src/emqx_mgmt_api_plugins.erl

```diff
@@ -565,6 +564,7 @@ install_package(FileName, Bin) ->
     ok = filelib:ensure_dir(File),
     ok = file:write_file(File, Bin),
     PackageName = string:trim(FileName, trailing, ".tar.gz"),
+    put(?fresh_install, true),
```

Comment:
why not an argument passed to emqx_plugins:ensure_installed?

Reply:
Installation from the management API(`?fresh_install` here) and when a node joins a cluster will execute similar code paths, such as creating plugin configuration directories, building schhema serde, etc.

The only difference is `emqx_plugins:ensure_plugin_config/1`
Installation when joining a cluster requires obtaining the existing configuration of the cluster, and a new installation will directly copy the default configuration file from tarball.
Passing parameters to the emqx_plugins:ensure_installed function will result in redundant function clause and `case...of`